File: world_commands.py
from __future__ import annotations
import json
import discord
from discord import app_commands
from discord.ext import commands

class WorldCog(commands.GroupCog, group_name="world", group_description="KIDDO schools, relationships, groups, routines and schedules"):
    def __init__(self, bot): self.bot = bot

    async def child_auto(self, interaction: discord.Interaction, current: str):
        rows = await self.bot.db.fetchall("SELECT name FROM children WHERE guild_id=? AND lower(name) LIKE lower(?) ORDER BY name LIMIT 25", (interaction.guild_id, f"%{current}%"))
        return [app_commands.Choice(name=r["name"], value=r["name"]) for r in rows]

    @app_commands.command(name="relationship", description="Add or update an important adult/family relationship")
    @app_commands.autocomplete(child=child_auto)
    async def relationship(self, interaction: discord.Interaction, child: str, person_name: str, relationship_type: str,
                           familiarity: app_commands.Range[int,0,100]=70, closeness: app_commands.Range[int,0,100]=70,
                           trust: app_commands.Range[int,0,100]=70, authority: app_commands.Range[int,0,100]=50, notes: str|None=None):
        c = await self.bot.db.get_child(interaction.guild_id, child)
        if not c: return await interaction.response.send_message("Child not found.", ephemeral=True)
        old = await self.bot.db.fetchone("SELECT id FROM relationships WHERE child_id=? AND lower(person_name)=lower(?)", (c["id"],person_name))
        if old:
            await self.bot.db.execute("UPDATE relationships SET relationship_type=?,familiarity=?,closeness=?,trust=?,authority=?,notes=? WHERE id=?", (relationship_type,familiarity,closeness,trust,authority,notes,old["id"]))
        else:
            await self.bot.db.execute("INSERT INTO relationships(child_id,person_name,relationship_type,familiarity,closeness,trust,authority,notes) VALUES(?,?,?,?,?,?,?,?)", (c["id"],person_name,relationship_type,familiarity,closeness,trust,authority,notes))
        await interaction.response.send_message(f"👥 Saved **{child} ↔ {person_name}** ({relationship_type}).")

    @app_commands.command(name="peer", description="Connect two registered children as friends, siblings, rivals, crushes, etc.")
    @app_commands.autocomplete(child_a=child_auto, child_b=child_auto)
    async def peer(self, interaction: discord.Interaction, child_a: str, child_b: str, relationship_type: str,
                   closeness: app_commands.Range[int,0,100]=50, conflict: app_commands.Range[int,0,100]=10,
                   jealousy: app_commands.Range[int,0,100]=0, protectiveness: app_commands.Range[int,0,100]=0,
                   dynamic_notes: str|None=None):
        a = await self.bot.db.get_child(interaction.guild_id, child_a); b = await self.bot.db.get_child(interaction.guild_id, child_b)
        if not a or not b: return await interaction.response.send_message("I couldn't find both registered children.", ephemeral=True)
        if a["id"] == b["id"]: return await interaction.response.send_message("Choose two different children.", ephemeral=True)
        x,y = sorted((a["id"],b["id"]))
        old = await self.bot.db.fetchone("SELECT id FROM child_relationships WHERE guild_id=? AND child_a=? AND child_b=?", (interaction.guild_id,x,y))
        vals=(relationship_type,closeness,conflict,jealousy,protectiveness,dynamic_notes)
        if old:
            await self.bot.db.execute("UPDATE child_relationships SET relationship_type=?,closeness=?,conflict=?,jealousy=?,protectiveness=?,dynamic_notes=? WHERE id=?", (*vals,old["id"]))
        else:
            await self.bot.db.execute("INSERT INTO child_relationships(guild_id,child_a,child_b,relationship_type,closeness,conflict,jealousy,protectiveness,dynamic_notes) VALUES(?,?,?,?,?,?,?,?,?)", (interaction.guild_id,x,y,*vals))
        await interaction.response.send_message(f"🧩 Saved **{child_a} ↔ {child_b}** as **{relationship_type}**.")

    @app_commands.command(name="school-create", description="Create a school, daycare or preschool")
    async def school_create(self, interaction: discord.Interaction, name: str, school_type: str, start_time: str|None=None, end_time: str|None=None, environment: str|None=None, notes: str|None=None):
        try:
            await self.bot.db.execute("INSERT INTO schools(guild_id,name,type,start_time,end_time,environment,notes) VALUES(?,?,?,?,?,?,?)", (interaction.guild_id,name,school_type,start_time,end_time,environment,notes))
        except Exception:
            return await interaction.response.send_message("That school already exists or couldn't be created.", ephemeral=True)
        await interaction.response.send_message(f"🏫 Created **{name}** ({school_type}).")

    @app_commands.command(name="classroom-create", description="Create a classroom within a registered school")
    async def classroom_create(self, interaction: discord.Interaction, school_name: str, classroom_name: str, grade: str|None=None, teacher_name: str|None=None, teacher_style: str|None=None, notes: str|None=None):
        s = await self.bot.db.fetchone("SELECT id FROM schools WHERE guild_id=? AND lower(name)=lower(?)", (interaction.guild_id,school_name))
        if not s: return await interaction.response.send_message("School not found.", ephemeral=True)
        await self.bot.db.execute("INSERT INTO classrooms(school_id,name,grade,teacher_name,teacher_style,notes) VALUES(?,?,?,?,?,?)", (s["id"],classroom_name,grade,teacher_name,teacher_style,notes))
        await interaction.response.send_message(f"📝 Created **{classroom_name}** at **{school_name}**.")

    @app_commands.command(name="enroll", description="Enroll a registered child in a classroom")
    @app_commands.autocomplete(child=child_auto)
    async def enroll(self, interaction: discord.Interaction, child: str, school_name: str, classroom_name: str):
        c = await self.bot.db.get_child(interaction.guild_id, child)
        if not c: return await interaction.response.send_message("Child not found.", ephemeral=True)
        room = await self.bot.db.fetchone("""SELECT c.id classroom_id,s.id school_id FROM classrooms c JOIN schools s ON s.id=c.school_id
            WHERE s.guild_id=? AND lower(s.name)=lower(?) AND lower(c.name)=lower(?)""", (interaction.guild_id,school_name,classroom_name))
        if not room: return await interaction.response.send_message("School/classroom not found.", ephemeral=True)
        await self.bot.db.execute("INSERT INTO enrollments(child_id,classroom_id) VALUES(?,?) ON CONFLICT(child_id) DO UPDATE SET classroom_id=excluded.classroom_id", (c["id"],room["classroom_id"]))
        await interaction.response.send_message(f"🎒 Enrolled **{child}** in **{classroom_name}** at **{school_name}**.")

    @app_commands.command(name="location-create", description="Create a custom place such as Nana's house or dance studio")
    async def location_create(self, interaction: discord.Interaction, name: str, location_type: str, environment: str|None=None, rules_style: str|None=None, notes: str|None=None):
        try:
            await self.bot.db.execute("INSERT INTO locations(guild_id,name,type,environment,rules_style,notes) VALUES(?,?,?,?,?,?)", (interaction.guild_id,name,location_type,environment,rules_style,notes))
        except Exception:
            return await interaction.response.send_message("That location already exists.", ephemeral=True)
        await interaction.response.send_message(f"📍 Created **{name}**.")

    @app_commands.command(name="schedule-add", description="Add a recurring weekly place/activity to a child's world")
    @app_commands.autocomplete(child=child_auto)
    async def schedule_add(self, interaction: discord.Interaction, child: str, weekday: app_commands.Range[int,0,6], location_name: str, start_time: str, end_time: str, notes: str|None=None):
        c=await self.bot.db.get_child(interaction.guild_id,child)
        if not c:return await interaction.response.send_message("Child not found.",ephemeral=True)
        await self.bot.db.execute("INSERT INTO schedules(child_id,weekday,location_name,start_time,end_time,notes) VALUES(?,?,?,?,?,?)",(c["id"],weekday,location_name,start_time,end_time,notes))
        days=["Mon","Tue","Wed","Thu","Fri","Sat","Sun"]
        await interaction.response.send_message(f"📅 **{child}**: {days[weekday]} {start_time}–{end_time} at **{location_name}**.")

    @app_commands.command(name="routine-set", description="Save a routine as ordered steps separated by |")
    @app_commands.autocomplete(child=child_auto)
    async def routine_set(self, interaction: discord.Interaction, child: str, routine_name: str, steps: str, notes: str|None=None):
        c=await self.bot.db.get_child(interaction.guild_id,child)
        if not c:return await interaction.response.send_message("Child not found.",ephemeral=True)
        step_list=[x.strip() for x in steps.split("|") if x.strip()]
        await self.bot.db.execute("INSERT INTO routines(child_id,name,steps_json,notes) VALUES(?,?,?,?) ON CONFLICT(child_id,name) DO UPDATE SET steps_json=excluded.steps_json,notes=excluded.notes",(c["id"],routine_name,json.dumps(step_list),notes))
        await interaction.response.send_message(f"🛁 Saved **{routine_name}** for **{child}** with {len(step_list)} steps.")

    @app_commands.command(name="preference", description="Confirm a child's like or dislike as canon")
    @app_commands.autocomplete(child=child_auto)
    async def preference(self, interaction: discord.Interaction, child: str, category: str, item: str, preference_type: str):
        if preference_type not in ("like","dislike"): return await interaction.response.send_message("preference_type must be `like` or `dislike`.",ephemeral=True)
        c=await self.bot.db.get_child(interaction.guild_id,child)
        if not c:return await interaction.response.send_message("Child not found.",ephemeral=True)
        await self.bot.db.execute("INSERT INTO preferences(child_id,category,item,preference_type,confidence,confirmed) VALUES(?,?,?,?,100,1) ON CONFLICT(child_id,category,item,preference_type) DO UPDATE SET confidence=100,confirmed=1",(c["id"],category,item,preference_type))
        await interaction.response.send_message(f"💛 Canon: **{child}** {preference_type}s **{item}** ({category}).")

    @app_commands.command(name="group-create", description="Create a friend/social group")
    async def group_create(self, interaction: discord.Interaction, name: str, group_type: str="Friend Group", notes: str|None=None):
        try: await self.bot.db.execute("INSERT INTO groups_tbl(guild_id,name,type,notes) VALUES(?,?,?,?)",(interaction.guild_id,name,group_type,notes))
        except Exception:return await interaction.response.send_message("That group already exists.",ephemeral=True)
        await interaction.response.send_message(f"👯 Created **{name}**.")

    @app_commands.command(name="group-add", description="Add a registered child to a social group")
    @app_commands.autocomplete(child=child_auto)
    async def group_add(self, interaction: discord.Interaction, group_name: str, child: str):
        g=await self.bot.db.fetchone("SELECT id FROM groups_tbl WHERE guild_id=? AND lower(name)=lower(?)",(interaction.guild_id,group_name)); c=await self.bot.db.get_child(interaction.guild_id,child)
        if not g or not c:return await interaction.response.send_message("Group or child not found.",ephemeral=True)
        await self.bot.db.execute("INSERT OR IGNORE INTO group_members(group_id,child_id) VALUES(?,?)",(g["id"],c["id"]))
        await interaction.response.send_message(f"➕ Added **{child}** to **{group_name}**.")

async def setup(bot): await bot.add_cog(WorldCog(bot))

File: worldCommands.js
import { SlashCommandBuilder } from "discord.js";

const world = new SlashCommandBuilder().setName("kiddo-world").setDescription("Manage KIDDO's shared RP world")
  .addSubcommand(s => s.setName("adult").setDescription("Save a child's relationship with an adult/caregiver")
    .addStringOption(o => o.setName("child").setDescription("Child name").setAutocomplete(true).setRequired(true))
    .addStringOption(o => o.setName("person").setDescription("Adult/caregiver name").setRequired(true))
    .addStringOption(o => o.setName("relationship").setDescription("Parent, Grandma, Nanny, etc.").setRequired(true))
    .addIntegerOption(o => o.setName("closeness").setDescription("0-100").setMinValue(0).setMaxValue(100))
    .addIntegerOption(o => o.setName("authority").setDescription("0-100").setMinValue(0).setMaxValue(100))
    .addStringOption(o => o.setName("notes").setDescription("Dynamic notes")))
  .addSubcommand(s => s.setName("peer").setDescription("Connect two registered children")
    .addStringOption(o => o.setName("child_a").setDescription("First child").setAutocomplete(true).setRequired(true))
    .addStringOption(o => o.setName("child_b").setDescription("Second child").setAutocomplete(true).setRequired(true))
    .addStringOption(o => o.setName("relationship").setDescription("Sibling, best friend, rival, crush, etc.").setRequired(true))
    .addIntegerOption(o => o.setName("closeness").setDescription("0-100").setMinValue(0).setMaxValue(100))
    .addIntegerOption(o => o.setName("conflict").setDescription("0-100").setMinValue(0).setMaxValue(100))
    .addIntegerOption(o => o.setName("jealousy").setDescription("0-100").setMinValue(0).setMaxValue(100))
    .addIntegerOption(o => o.setName("protectiveness").setDescription("0-100").setMinValue(0).setMaxValue(100))
    .addStringOption(o => o.setName("notes").setDescription("Dynamic notes")))
  .addSubcommand(s => s.setName("school-create").setDescription("Create a school/daycare/preschool")
    .addStringOption(o => o.setName("name").setDescription("School name").setRequired(true))
    .addStringOption(o => o.setName("type").setDescription("Daycare, preschool, elementary, etc.").setRequired(true))
    .addStringOption(o => o.setName("environment").setDescription("School vibe/environment"))
    .addStringOption(o => o.setName("notes").setDescription("Notes")))
  .addSubcommand(s => s.setName("classroom-create").setDescription("Create a classroom")
    .addStringOption(o => o.setName("school").setDescription("School name").setRequired(true))
    .addStringOption(o => o.setName("classroom").setDescription("Classroom name").setRequired(true))
    .addStringOption(o => o.setName("grade").setDescription("Grade"))
    .addStringOption(o => o.setName("teacher").setDescription("Teacher name"))
    .addStringOption(o => o.setName("teacher_style").setDescription("Warm, strict, playful, etc.")))
  .addSubcommand(s => s.setName("enroll").setDescription("Enroll a registered child")
    .addStringOption(o => o.setName("child").setDescription("Child name").setAutocomplete(true).setRequired(true))
    .addStringOption(o => o.setName("school").setDescription("School name").setRequired(true))
    .addStringOption(o => o.setName("classroom").setDescription("Classroom name").setRequired(true)))
  .addSubcommand(s => s.setName("location-create").setDescription("Create a custom place")
    .addStringOption(o => o.setName("name").setDescription("Location name").setRequired(true))
    .addStringOption(o => o.setName("type").setDescription("Grandparent house, studio, etc.").setRequired(true))
    .addStringOption(o => o.setName("environment").setDescription("Environment"))
    .addStringOption(o => o.setName("rules").setDescription("Strict, lenient, structured, etc."))
    .addStringOption(o => o.setName("notes").setDescription("Notes")))
  .addSubcommand(s => s.setName("schedule-add").setDescription("Add a recurring weekly activity")
    .addStringOption(o => o.setName("child").setDescription("Child name").setAutocomplete(true).setRequired(true))
    .addIntegerOption(o => o.setName("weekday").setDescription("0=Mon through 6=Sun").setMinValue(0).setMaxValue(6).setRequired(true))
    .addStringOption(o => o.setName("location").setDescription("Place/activity").setRequired(true))
    .addStringOption(o => o.setName("start").setDescription("Start time").setRequired(true))
    .addStringOption(o => o.setName("end").setDescription("End time").setRequired(true))
    .addStringOption(o => o.setName("notes").setDescription("Notes")))
  .addSubcommand(s => s.setName("routine-set").setDescription("Save a routine; separate steps with |")
    .addStringOption(o => o.setName("child").setDescription("Child name").setAutocomplete(true).setRequired(true))
    .addStringOption(o => o.setName("name").setDescription("Routine name").setRequired(true))
    .addStringOption(o => o.setName("steps").setDescription("Bath | pajamas | story | bed").setRequired(true))
    .addStringOption(o => o.setName("notes").setDescription("Notes")))
  .addSubcommand(s => s.setName("preference").setDescription("Confirm a like/dislike as canon")
    .addStringOption(o => o.setName("child").setDescription("Child name").setAutocomplete(true).setRequired(true))
    .addStringOption(o => o.setName("category").setDescription("Food, music, toy, subject, etc.").setRequired(true))
    .addStringOption(o => o.setName("item").setDescription("Preference item").setRequired(true))
    .addStringOption(o => o.setName("type").setDescription("Like or dislike").setRequired(true).addChoices({ name: "Like", value: "like" }, { name: "Dislike", value: "dislike" })))
  .addSubcommand(s => s.setName("group-create").setDescription("Create a social/friend group")
    .addStringOption(o => o.setName("name").setDescription("Group name").setRequired(true))
    .addStringOption(o => o.setName("type").setDescription("Friend Group, Lunch Table, etc."))
    .addStringOption(o => o.setName("notes").setDescription("Notes")))
  .addSubcommand(s => s.setName("group-add").setDescription("Add a child to a group")
    .addStringOption(o => o.setName("group").setDescription("Group name").setRequired(true))
    .addStringOption(o => o.setName("child").setDescription("Child name").setAutocomplete(true).setRequired(true)));

export const worldCommand = world;

export async function handleWorld(interaction, db) {
  const sub = interaction.options.getSubcommand();
  const getChild = name => db.getChild(interaction.guildId, name);

  if (sub === "adult") {
    const c = getChild(interaction.options.getString("child", true)); if (!c) return interaction.reply({ content: "Child not found.", ephemeral: true });
    const person = interaction.options.getString("person", true); const rel = interaction.options.getString("relationship", true);
    const close = interaction.options.getInteger("closeness") ?? 50, auth = interaction.options.getInteger("authority") ?? 50, notes = interaction.options.getString("notes");
    const old = db.get("SELECT id FROM relationships WHERE child_id=? AND lower(person_name)=lower(?)", [c.id, person]);
    if (old) db.run("UPDATE relationships SET relationship_type=?,closeness=?,authority=?,notes=? WHERE id=?", [rel, close, auth, notes, old.id]);
    else db.run("INSERT INTO relationships(child_id,person_name,relationship_type,closeness,authority,notes) VALUES(?,?,?,?,?,?)", [c.id, person, rel, close, auth, notes]);
    return interaction.reply(`👥 Saved **${c.name} ↔ ${person}** (${rel}).`);
  }
  if (sub === "peer") {
    const a = getChild(interaction.options.getString("child_a", true)), b = getChild(interaction.options.getString("child_b", true));
    if (!a || !b || a.id === b.id) return interaction.reply({ content: "Choose two different registered children.", ephemeral: true });
    const [x, y] = [a.id, b.id].sort((m,n) => m-n);
    const vals = [interaction.options.getString("relationship", true), interaction.options.getInteger("closeness") ?? 50, interaction.options.getInteger("conflict") ?? 10, interaction.options.getInteger("jealousy") ?? 0, interaction.options.getInteger("protectiveness") ?? 0, interaction.options.getString("notes")];
    const old = db.get("SELECT id FROM child_relationships WHERE guild_id=? AND child_a=? AND child_b=?", [interaction.guildId,x,y]);
    if (old) db.run("UPDATE child_relationships SET relationship_type=?,closeness=?,conflict=?,jealousy=?,protectiveness=?,dynamic_notes=? WHERE id=?", [...vals, old.id]);
    else db.run("INSERT INTO child_relationships(guild_id,child_a,child_b,relationship_type,closeness,conflict,jealousy,protectiveness,dynamic_notes) VALUES(?,?,?,?,?,?,?,?,?)", [interaction.guildId,x,y,...vals]);
    return interaction.reply(`🧩 Saved **${a.name} ↔ ${b.name}** as **${vals[0]}**.`);
  }
  if (sub === "school-create") {
    try { db.run("INSERT INTO schools(guild_id,name,type,environment,notes) VALUES(?,?,?,?,?)", [interaction.guildId, interaction.options.getString("name",true), interaction.options.getString("type",true), interaction.options.getString("environment"), interaction.options.getString("notes")]); }
    catch { return interaction.reply({ content: "That school already exists or couldn't be created.", ephemeral: true }); }
    return interaction.reply("🏫 School created.");
  }
  if (sub === "classroom-create") {
    const school = db.get("SELECT id FROM schools WHERE guild_id=? AND lower(name)=lower(?)", [interaction.guildId, interaction.options.getString("school",true)]);
    if (!school) return interaction.reply({ content: "School not found.", ephemeral: true });
    db.run("INSERT INTO classrooms(school_id,name,grade,teacher_name,teacher_style) VALUES(?,?,?,?,?)", [school.id, interaction.options.getString("classroom",true), interaction.options.getString("grade"), interaction.options.getString("teacher"), interaction.options.getString("teacher_style")]);
    return interaction.reply("📝 Classroom created.");
  }
  if (sub === "enroll") {
    const c = getChild(interaction.options.getString("child",true)); if (!c) return interaction.reply({ content: "Child not found.", ephemeral:true });
    const room = db.get(`SELECT c.id classroom_id FROM classrooms c JOIN schools s ON s.id=c.school_id WHERE s.guild_id=? AND lower(s.name)=lower(?) AND lower(c.name)=lower(?)`, [interaction.guildId, interaction.options.getString("school",true), interaction.options.getString("classroom",true)]);
    if (!room) return interaction.reply({ content: "School/classroom not found.", ephemeral:true });
    db.run("INSERT INTO enrollments(child_id,classroom_id) VALUES(?,?) ON CONFLICT(child_id) DO UPDATE SET classroom_id=excluded.classroom_id", [c.id, room.classroom_id]);
    return interaction.reply(`🎒 Enrolled **${c.name}**.`);
  }
  if (sub === "location-create") {
    try { db.run("INSERT INTO locations(guild_id,name,type,environment,rules_style,notes) VALUES(?,?,?,?,?,?)", [interaction.guildId, interaction.options.getString("name",true), interaction.options.getString("type",true), interaction.options.getString("environment"), interaction.options.getString("rules"), interaction.options.getString("notes")]); }
    catch { return interaction.reply({ content:"That location already exists.", ephemeral:true }); }
    return interaction.reply("📍 Location created.");
  }
  if (sub === "schedule-add") {
    const c = getChild(interaction.options.getString("child",true)); if (!c) return interaction.reply({content:"Child not found.",ephemeral:true});
    db.run("INSERT INTO schedules(child_id,weekday,location_name,start_time,end_time,notes) VALUES(?,?,?,?,?,?)", [c.id, interaction.options.getInteger("weekday",true), interaction.options.getString("location",true), interaction.options.getString("start",true), interaction.options.getString("end",true), interaction.options.getString("notes")]);
    return interaction.reply(`📅 Added to **${c.name}**'s schedule.`);
  }
  if (sub === "routine-set") {
    const c = getChild(interaction.options.getString("child",true)); if (!c) return interaction.reply({content:"Child not found.",ephemeral:true});
    const steps = interaction.options.getString("steps",true).split("|").map(x=>x.trim()).filter(Boolean);
    db.run("INSERT INTO routines(child_id,name,steps_json,notes) VALUES(?,?,?,?) ON CONFLICT(child_id,name) DO UPDATE SET steps_json=excluded.steps_json,notes=excluded.notes", [c.id, interaction.options.getString("name",true), JSON.stringify(steps), interaction.options.getString("notes")]);
    return interaction.reply(`🛁 Routine saved for **${c.name}**.`);
  }
  if (sub === "preference") {
    const c = getChild(interaction.options.getString("child",true)); if (!c) return interaction.reply({content:"Child not found.",ephemeral:true});
    db.run("INSERT INTO preferences(child_id,category,item,preference_type,confidence,confirmed) VALUES(?,?,?,?,100,1) ON CONFLICT(child_id,category,item,preference_type) DO UPDATE SET confidence=100,confirmed=1", [c.id, interaction.options.getString("category",true), interaction.options.getString("item",true), interaction.options.getString("type",true)]);
    return interaction.reply(`💛 Canon preference saved for **${c.name}**.`);
  }
  if (sub === "group-create") {
    try { db.run("INSERT INTO groups_tbl(guild_id,name,type,notes) VALUES(?,?,?,?)", [interaction.guildId, interaction.options.getString("name",true), interaction.options.getString("type") || "Friend Group", interaction.options.getString("notes")]); }
    catch { return interaction.reply({content:"That group already exists.",ephemeral:true}); }
    return interaction.reply("👯 Group created.");
  }
  if (sub === "group-add") {
    const c = getChild(interaction.options.getString("child",true)), g = db.get("SELECT id FROM groups_tbl WHERE guild_id=? AND lower(name)=lower(?)", [interaction.guildId, interaction.options.getString("group",true)]);
    if (!c || !g) return interaction.reply({content:"Group or child not found.",ephemeral:true});
    db.run("INSERT OR IGNORE INTO group_members(group_id,child_id) VALUES(?,?)", [g.id,c.id]);
    return interaction.reply(`➕ Added **${c.name}** to the group.`);
  }
}
